projects/core/src/cart/facade/multi-cart.service.ts

```diff
@@ -9,6 +9,9 @@ import { CartActions } from '../store/actions/index';
 import { StateWithMultiCart } from '../store/multi-cart-state';
 import { MultiCartSelectors } from '../store/selectors/index';
 
+const TWO = 2;
+const NINE = 9;
```

Comment:
Please try don't define variables in whole file scope, you can add just
```const RADIX = 36, FROM = 2, LENGTH = 9;```
just inside method `generateTempCartId`

please apply that comment in other places